awx/ui/client/features/output/details.component.js

```diff
@@ -735,11 +748,12 @@ function JobDetailsController (
         vm.toggleLabels = toggleLabels;
         vm.showLabels = showLabels;
 
-        unsubscribe = subscribe(({ status, started, finished, scm }) => {
+        unsubscribe = subscribe(({ status, started, finished, scm, environment }) => {
```

Comment:
this is the right place to add this. 

for `environment` to be subscribe-able we'll need to also add some lines to `status.service.js`:

Mainly here: https://github.com/ansible/awx/blob/devel/awx/ui/client/features/output/status.service.js#L299 (if we want to use our end-of-job sync-up hack)

an additional `this.state` variable for the virtualenv would need to be added as well: https://github.com/ansible/awx/blob/devel/awx/ui/client/features/output/status.service.js#L29

we'd want to initialize that state variable to whatever is defined on the model. Unless we actually expect to receive this data over websockets, the only place that variable would need to be updated would be in the `sync` function.

Reply:
Note that if we're going with the sync function to "fake" event data updates (which is OK IMHO), the job would still need to at least emit an `EOF` event for this data to show without a refresh.